src/pybind/mgr/dashboard/frontend/src/app/shared/components/modal/modal.component.ts

```diff
@@ -0,0 +1,19 @@
+import { Component, Input, TemplateRef } from '@angular/core';
+
+import { BsModalService } from 'ngx-bootstrap/modal';
+import { BsModalRef } from 'ngx-bootstrap/modal/bs-modal-ref.service';
+
+@Component({
+  selector: 'cd-modal',
+  templateUrl: './modal.component.html',
+  styleUrls: ['./modal.component.scss']
+})
+export class ModalComponent {
+  @Input() modalRef: BsModalRef;
+
+  constructor(private modalService: BsModalService) {}
+
+  openModal(template: TemplateRef<any>) {
```

Comment:
Consider renaming this method to ``showModal``.

Reply:
done